builtin/nomad/jobspec/platform.go

```diff
@@ -46,86 +54,314 @@ func (p *Platform) GenerationFunc() interface{} {
 	return p.Generation
 }
 
-// Deploy deploys an image to Nomad.
-func (p *Platform) Deploy(
-	ctx context.Context,
-	log hclog.Logger,
-	src *component.Source,
-	img *docker.Image,
-	deployConfig *component.DeploymentConfig,
-	ui terminal.UI,
-) (*nomad.Deployment, error) {
-	// We'll update the user in real time
-	sg := ui.StepGroup()
-	defer sg.Wait()
-	s := sg.Add("Initializing the Nomad client...")
-	defer func() { s.Abort() }()
+// StatusFunc implements component.Status
+func (p *Platform) StatusFunc() interface{} {
+	return p.Status
+}
 
+func (p *Platform) resourceManager(log hclog.Logger, dcr *component.DeclaredResourcesResp) *resource.Manager {
+	return resource.NewManager(
+		resource.WithLogger(log.Named("resource_manager")),
+		resource.WithValueProvider(p.getNomadClient),
+		resource.WithDeclaredResourcesResp(dcr),
+		resource.WithResource(resource.NewResource(
+			resource.WithName(rmResourceJobName),
+			resource.WithState(&Resource_Job{}),
+			resource.WithCreate(p.resourceJobCreate),
+			resource.WithDestroy(p.resourceJobDestroy),
+			resource.WithStatus(p.resourceJobStatus),
+			resource.WithPlatform("jobspec"),
+		)),
+	)
+}
+
+// getNomadJobspecClient is a value provider for our resource manager and provides
+// the client connection used by resources to interact with Nomad.
+func (p *Platform) getNomadClient() (*nomadClient, error) {
 	// Get our client
 	client, err := api.NewClient(api.DefaultConfig())
 	if err != nil {
 		return nil, err
 	}
-	jobclient := client.Jobs()
+	return &nomadClient{
+		NomadClient: client,
+	}, nil
+}
+
+func (p *Platform) resourceJobCreate(
+	ctx context.Context,
+	client *nomadClient,
+	result *Deployment,
+	deployConfig *component.DeploymentConfig,
+	img *docker.Image,
+	ui terminal.UI,
+	state *Resource_Job,
+) error {
+	st := ui.Status()
+	defer st.Close()
+	jobclient := client.NomadClient.Jobs()
 
 	// Parse the HCL
-	s.Update("Parsing the job specification...")
-	job, err := p.jobspec(client, p.config.Jobspec)
+	st.Update("Parsing the job specification...")
+	job, err := p.jobspec(client.NomadClient, p.config.Jobspec)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	// Create our deployment and set an initial ID
-	var result nomad.Deployment
 	result.Id = deployConfig.Id
 	result.Name = *job.ID
 
 	// Set our deployment ID on the meta.
 	job.SetMeta(metaId, result.Id)
 
 	// Update our client to use the Namespace set in the jobspec
-	client.SetNamespace(*job.Namespace)
+	client.NomadClient.SetNamespace(*job.Namespace)
 
 	// Register job
-	s.Update("Registering job %q...", *job.Name)
+	st.Update(fmt.Sprintf("Registering job %q...", *job.Name))
```

Comment:
Same feedback here, no need for `fmt`, the StepGroup func should support string formatting!